content/documentation/master/schema.md

```diff
@@ -629,6 +603,88 @@ Subscriptions allow you to perform a query and whenever a backing object for tha
     subscription foo {
         # normal graphql query
     }
+
 {{< / highlight >}}
 
 See the page on [subscriptions](../subscriptions) for more details
+
+# Changing Schema
+
+The `GraphQLSchema` is an immutable object once its is built. To make things more complicated it is in fact an immutable
+directed acyclic graph (DAG).
```

Comment:
the schema itself is a general graph, not necessarily acyclic. 

The immutable parts are a DAG (because there is no cycles with pure immutable Objects in Java), but we have type references which gets resolved (and are the only not immutable parts). After they got resolved we a general graph, not a DAG.

Reply:
fixed